pkg/log/context.go

```diff
@@ -74,40 +94,70 @@ func (s *Settings) Validate() error {
 	if len(s.Format) == 0 {
 		return fmt.Errorf("validate Settings: LogFormat missing")
 	}
-	if s.Output == nil {
+	if len(s.Output) == 0 {
 		return fmt.Errorf("validate Settings: LogOutput missing")
 	}
+
 	return nil
 }
 
 // Configure creates a new context with a logger using the provided settings.
 func Configure(ctx context.Context, settings *Settings) context.Context {
-	once.Do(func() {
-		level, err := logrus.ParseLevel(settings.Level)
-		if err != nil {
-			panic(fmt.Sprintf("Could not parse log level configuration: %s", err))
-		}
-		formatter, ok := supportedFormatters[settings.Format]
-		if !ok {
-			panic(fmt.Sprintf("Invalid log format: %s", settings.Format))
-		}
-		logger := &logrus.Logger{
-			Formatter: formatter,
-			Level:     level,
-			Out:       settings.Output,
-			Hooks:     make(logrus.LevelHooks),
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	currentSettings = settings
+	level, err := logrus.ParseLevel(settings.Level)
+	if err != nil {
+		panic(fmt.Sprintf("Could not parse log level configuration: %s", err))
```

Comment:
As this is now called at runtime, I think panicking should be replaced with returning an error. Otherwise you'll get 500 when providing `level: dummy` instead of 400.